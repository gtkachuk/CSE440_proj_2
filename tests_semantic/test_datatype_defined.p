program testDataTypeDefined;

class myCustomClass
BEGIN
   VAR
      test : integer;
FUNCTION myCustomClass;
BEGIN
   test := 1
END   
END



class myCustomClass
BEGIN
   VAR
      test : integer;
FUNCTION myCustomClass;
BEGIN
   test := 2
END   
END



class testDataTypeDefined 
BEGIN
   VAR 
      lightSwitch	      : myCustomClass;
      compilerWorks	      : integer;

FUNCTION TestDataTypeDefined;
BEGIN
   compilerWorks := 1
END

END
.

