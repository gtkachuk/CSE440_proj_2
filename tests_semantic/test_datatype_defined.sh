#!/bin/sh

. ../tests_lib.inc

test_diff_sem test_datatype_defined.p
