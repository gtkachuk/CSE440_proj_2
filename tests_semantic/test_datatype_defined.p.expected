ERROR:16:Class "myCustomClass" already declared at line 4
Errors detected. Exiting.
